clang/include/clang/Basic/arm_neon.td

```diff
@@ -1354,9 +1354,9 @@ let isScalarNarrowShift = 1 in {
   // Signed/Unsigned Saturating Rounded Shift Right Narrow (Immediate)
   def SCALAR_SQRSHRN_N: SInst<"vqrshrn_n", "(1<)1I", "SsSiSlSUsSUiSUl">;
   // Signed Saturating Shift Right Unsigned Narrow (Immediate)
-  def SCALAR_SQSHRUN_N: SInst<"vqshrun_n", "(1<)1I", "SsSiSl">;
+  def SCALAR_SQSHRUN_N: SInst<"vqshrun_n", "(1<U)1I", "SsSiSl">;
```

Comment:
You have updated this in tablegen, have we covered all the test cases in `clang/test/CodeGen/aarch64-neon-intrinsics.c`?

From a quick grep, it looks like there are some test cases that need their integer type changing from signed to unsigned.